snippets/zoe/contracts/test-loan.js

```diff
@@ -0,0 +1,141 @@
+// @ts-check
+
+import '@agoric/install-ses';
+import { makeFakeVatAdmin } from '@agoric/zoe/src/contractFacet/fakeVatAdmin';
+import { makeZoe } from '@agoric/zoe';
+import bundleSource from '@agoric/bundle-source';
+import { makeIssuerKit } from '@agoric/ertp';
+import test from 'ava';
+import { E } from '@agoric/eventual-send';
+import buildManualTimer from '@agoric/zoe/tools/manualTimer';
+import { makeFakePriceAuthority } from '@agoric/zoe/tools/fakePriceAuthority';
+import { makeSubscriptionKit } from '@agoric/notifier';
+
+test('loan contract', async t => {
+  const zoe = makeZoe(makeFakeVatAdmin().admin);
+
+  const contractBundle = await bundleSource(
+    require.resolve('@agoric/zoe/src/contracts/loan'),
+  );
+  const installation = await E(zoe).install(contractBundle);
+
+  const collateralKit = makeIssuerKit('moola');
+  const loanKit = makeIssuerKit('simoleans');
+
+  // Create autoswap installation and instance
+  const autoswapBundle = await bundleSource(
+    require.resolve('@agoric/zoe/src/contracts/autoswap'),
+  );
+  const autoswapInstallation = await E(zoe).install(autoswapBundle);
+
+  const { instance: autoswapInstance } = await E(zoe).startInstance(
+    autoswapInstallation,
+    harden({ Central: collateralKit.issuer, Secondary: loanKit.issuer }),
+  );
+
+  const issuerKeywordRecord = harden({
+    Collateral: collateralKit.issuer,
+    Loan: loanKit.issuer,
+  });
+
+  const timer = buildManualTimer(console.log);
+
+  const priceAuthority = makeFakePriceAuthority({
+    mathIn: collateralKit.amountMath,
+    mathOut: loanKit.amountMath,
+    priceList: [4, 2],
+    timer,
+  });
+
+  const doAddCollateral = _ => {};
+  const allCollateralAmount = collateralKit.amountMath.make(1000);
+  const x = loanKit.amountMath.make(1500);
```

Comment:
I think this would be more understandable and require less explanation if `x` were renamed to `myWarningLevel`.

Reply:
sounds good!